03-rnaseq/dimension-reduction_rnaseq_02_umap.Rmd

```diff
@@ -0,0 +1,414 @@
+We are using the default parameters when we run the `umap::umap()` function.
+Take a look at the `Tuning UMAP` section at [this link](https://cran.r-project.org/web/packages/umap/vignettes/umap.html) for more on parameters that can be configured when executing `umap::umap()` [@umap-parameters].
+You can also run the following in the RStudio console to get more information on the function and its default parameters: `?umap::umap` or `?umap::umap.defaults`.
+
+```{r}
+# First we are going to retrieve the normalized data from the `DESeqDataSet` object using the `assay()` function
+normalized_counts <- assay(dds_norm) %>%
+  t() # We need to transpose this data in preparation for the `umap()` function
+
+# Now let's tell R to perform UMAP on the normalized data
+umap_results <- umap::umap(normalized_counts)
+```
+
+## Prepare data frame for plotting
+
+Now that we have the results from UMAP, we need to extract the counts data from the `umap_results` object and merge the variables from the metadata that we will use for annotating our plot.
+
+```{r}
+# We need to tell R to create a data frame with the umap counts and annotation data in preparation for plotting with `ggplot2`
+umap_plot_df <- data.frame(umap_results$layout) %>%
+  tibble::rownames_to_column("refinebio_accession_code") %>% # Let's get the rownames into a column so we can merge the annotation data
+  dplyr::inner_join(metadata, by = "refinebio_accession_code") # Let's merge the annotation data using the `refinebio_accession_code` column
+```
+
+## Create UMAP plot
+
+Now we can use the `ggplot` function to plot our normalized UMAP scores.
+
+```{r}
+# Plot using `ggplot()` function
+ggplot(umap_plot_df, aes(x = X1, y = X2)) +
+  geom_point() # This tells R that we want a scatterplot
+```
+
+Let's try adding a variable to our plot for annotation.
+
+In this code chunk, the variable `refinebio_treatment` is given to the `ggplot()` function as part of the goal of the experiment is to analyze the sample transcriptional responses to androgen deprivation therapy (ADT).
+
+```{r}
+# Plot using `ggplot()` function
+ggplot(
+  umap_plot_df,
+  aes(
+    x = X1,
+    y = X2,
+    color = refinebio_treatment # Here we are giving `refinebio_treatment` to the color argument
+  )
+) +
+  geom_point() # This tells R that we want a scatterplot
+```
+
+In the next code chunk, we are going to add another variable to our plot for annotation.
+
+We'll plot using both `refinebio_treatment` and `refinebio_disease` variables for labels since they are central to the androgen deprivation therapy (ADT) based hypothesis in the [original paper](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6210624/) [@ADT-Prostate-Cancer].
+
+```{r}
+# Plot using `ggplot()` function
+final_annotated_umap_plot <- ggplot(
+  umap_plot_df,
+  aes(
+    x = X1,
+    y = X2,
+    color = refinebio_treatment,
+    # Here we are giving `refinebio_treatment` to the color argument
```

Comment:
Here I have the same comment from before: https://github.com/AlexsLemonade/refinebio-examples/pull/176#discussion_r474652655

Reply:
My guess is you had worked on this before we got to that point on the PCA notebook. 😃